Makefile

```diff
@@ -10,3 +10,32 @@ all:
 .PHONY: clean
 clean:
 	@-go clean -i
+
+.PHONY: ci
+ci: check-gofmt check-goimports check-golint vet test test-race
+
+# goimports doesn't support the -s flag to simplify code, therefore we use both
+# goimports and gofmt -s.
+.PHONY: check-gofmt
+check-gofmt:
+	bash -c 'diff <(gofmt -s -d .) <(printf "")'
+
+.PHONY: check-goimports
+check-goimports:
+	bash -c 'diff <(goimports -d .) <(printf "")'
+
+.PHONY: check-golint
+check-golint:
+	bash -c 'diff <(golint ./...) <(printf "")'
```

Comment:
Yeah, so far it does :)

I learned the diff trick from the internets... tested it, works as expected (to detect actual problems).